redhat-developer/devspaces
# chore: add a param to specify openvsx url

Signed-off-by: Valeriy Svydenko <vsvydenk@redhat.com>



### What does this PR do?
These changes are related to the customization of  a local plugin registry build.

Add  `openvsx-url` parameter to the build command to customize openvsx-server that is going to be used as a local VSX registry server.

If `openvsx-url` parameter was set, the build process will use `curl` command to download an archive with sources.  By default it uses sources from https://github.com/che-incubator/che-openvsx.git


To test changes, you can follow:

1. Start a workspace using https://devspaces.apps.sandbox-stage.gb17.p1.openshiftapps.com/dashboard/#/https://github.com/redhat-developer/devspaces/tree/sv-custom-openvsx-param
2. Open a terminal and go to `dependencies/che-plugin-registry`
3. Execute `./build.sh --openvsx-url https://github.com/svor/omnisharp-theia-plugin/releases/download/v0.0.5/che-openvsx.tar.gz`
4. The build should be successful


### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-5290

#### Release Notes

N/A

#### Docs PR (if applicable)
